Initial public draft of governance doc

Mierdin opened the pull request:
This is the initial draft for the NRE Labs governance document. Please use the "files changed" tab to inspect the contents and provide in-line comments where relevant.

The credit for this draft goes to @LCaywood - she asked that I post this via PR on her behalf.

Mierdin committed:
Initial public draft of governance doc

Signed-off-by: Matt Oswalt <matt@keepingitclassless.net>

LCaywood commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+
+Elections and any other voting decisions require a quorum of 50% of voting members. Voting can be held during Committee meetings, or by electronic means within a specified time period.
+
+Amendments to this or Policy documents require a majority vote.
```
...of the Steering Committee.

smk4664 commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+
+The Marketing Lead will be elected according to the same criteria and process as the Director. The Marketing Lead will have the following responsibilities.
+
+- Develop the Marketing portion, including metris, of the annual operating plan in consultation with the Technical Community. Report regularly to the Steering Committee on progress against plan.
```
"including metris" should be "including metrics".

LCaywood replied:
metrics, not metris

Mierdin committed:
Fixed minor typos

Signed-off-by: Matt Oswalt <matt@keepingitclassless.net>

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+Contributors will comply with the policies adopted by the Steering Committee of NRE Labs. These policies may include:
+
+- Contributor License Agreement
+- Code of Conduct
```
TODO: Add link to the CoC doc

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+
+Title to all trade or service marks used by the Project (“Project Trademarks”) is currently held by Juniper Networks. In the likely event that the Project joins an open source foundation or fiscal conservancy, Project Trademarks shall be transferred and assigned to that entity to hold on behalf of the Project. Any use of any Project Trademarks by participants in the Project or others shall be in  ...
+
+The Project shall develop and own all Project repositories and social media accounts, and domain name registrations created by the Project. The Platform is licensed under the Apache v2.0 license. Lessons contributed to the Project are published under CC BY 2.0. Project documentation, web and marketing content shall be published under Creative Commons license. 
```
I'd prefer to see some clear Copyright + License heading rather than a sentence inside that general section: these are probably the most important legal elements that contributors or users (reusers) should be concerned about IMHO (like, "can I copy/paste it to reuse it in class ?", or "can I fork the Antidote/Syringe Go code to create another SaaS startup ?"...)

LCaywood replied:
Will moved the license info further up in the doc. There may be a good call for an FAQ describing some of these things in more detail, but not needed in a governance doc as such.

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+## Mission & Scope
```
I'd welcome an introductory section to describe the purpose of that very document, and who should be concerned by reading it: mission of the document vs mission of the project ;-)

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+## Mission & Scope
+
+NRE Labs grew out of the experiences of working network engineers who were quick to see the value of embracing infrastructure automation for the purpose of network reliability, but were challenged by the learning curve of doing so.
```
TODO: Expand the NRE acronym ?

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+## Mission & Scope
+
```
Maybe a subheading "History", or similar here : or "Where did it came from". This one is rather opinionated and not likely to change, whereas the "mission" later may evolve in the future ?

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+We resolve to provide a platform for education in infrastructure automation tools and techniques for network engineers that can immediately be applied to real-world daily tasks.
+
+We encourage lessons of all types relating to network reliability engineering and infrastructure automation. All lessons should be concise and practical, focused on the mastery of a specific automation task.
+
```
Question : should the site/lessons limit to education about publicly available (i.e. open source) tools and techniques or should it be broader (incl. proprietary products) ?

LCaywood replied:
FAQ - clarify in contribution docs

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+
+### Mission
+
+NRE Labs has been an open source project from its inception--both the platform itself and all the lessons. The NRE Labs community seeks to ensure that anyone anywhere can quickly learn key automation concepts using whatever network equipment and tools they have available. Thus we encourage contributions from as broad a swath of the networking and automation communities as possible.  Anyone can co ...
```
The last sentence is a bit problematic : anyone could contribute to that governance document in principle, but not as easimy as contributing a lesson. One needs to climb a few steps inside the community in between. While the intent is valuable, I feel it's a bit too extensive in reality ;-)

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+
+### Scope
+
+The scope of the Project includes software development under an OSI-approved open source licence supporting the mission, including documentation, testing, integration and the creation of other artifacts that aid the development, deployment, operation or adoption of the open source software project.
```
That first sentence seems rather a template from a generic document: may need to be adapted: "The scope of NRE Labs...", and the repetition of " the open source software project" at the end is a bit redundant : just "the project" : lessons aren't typically open source software in some sense (hence different licences, etc.) so... not a big deal anyway.

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+
+The scope of the Project includes software development under an OSI-approved open source licence supporting the mission, including documentation, testing, integration and the creation of other artifacts that aid the development, deployment, operation or adoption of the open source software project.
+
+NRE Labs has two main components: the Platform (“Antidote”), and the Lessons. The Platform can serve a wide variety of purposes as a lesson delivery platform, and similarly, Lessons may be written for a wide variety of subjects which do not directly pertain to the Mission of the project. 
```
**currently** has two main components.

Maybe make it explicit here that the Platform is open source software, to make it explicit that you may wish to operate your own Antidote if you wish, and still be welcome in the Project ?

olberger replied:
In general, I fear the articulation between the 2 main components is something that's the most complex originality here : thinking about the codebase of Wordpress vs wordpress.com for instance. What if I want to contribute to Antidote platform something that has no value to NRE Labs' curriculum... I fear some roadmap tension here...

Not sure a separate governance for the Platform software, and another for the site's curriculum would be easier.

I'm just playing the devil's advocate, and overengineering ;-)

LCaywood replied:
Not helpful in a governance doc, which discusses how the project is to be run, but something to be included in a contributor FAQ.

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+
+The Technical Community will be responsible for all aspects of oversight relating to coordinating the technical direction of the Project, including the architecture and projects needed to achieve the Mission and Scope of the Project; for approving project or system proposals (including, but not limited to, incubation, deprecation, and changes to a sub-project’s scope) in accordance with the Contr ...
+
+Technical participation in the Project is open to anyone so long as they abide by the terms of this document and documents referenced herein. Technical Community meetings are conducted in English and are open to the public and recorded. The Technical Community generally will involve Contributors and Committers. Contributors include anyone in the Technical Community who contributes code, documenta ...
```
Not sure recording of the TC meetings should be mandated upfront in this doc. Documented at least, maybe ? Yay for transparency, but should all verbatims be systematically made available ?

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+
+### Steering Committee
+
+The Steering Committee is responsible for setting the strategic direction of the Project in accordance with its Mission. It provides approval and oversight of the Project’s budget and disbursements, Policy documents and other legal matters, marketing, and other non-technical activities of the Project. Committee business will be conducted in English and will be recorded for public access. 
```
Again, recording (as in Youtube videos) seems a bit too strong a requirement : discussing confidential matters like potential sponsors, exclusion of people, etc. Such should probably be conducted in private channels until agreed on, for legal reasons, so I doubt recordings could be made, in practice. I'd suggest to only mandate documentation/minutes of the meetings, allowing to exclude confidential elements from the records.

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+
+Contributors will comply with the policies adopted by the Steering Committee of NRE Labs. These policies may include:
+
+- Contributor License Agreement
```
Is there a proposed template for that one ? Does that mean that the Project should hold the copyright for all contributions, then ?

LCaywood replied:
We've been advised to have contributors use the DCO https://developercertificate.org/ vs the heavier CLA. This will be included in the contributor documentation.

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+
+- Contributor License Agreement
+- Code of Conduct
+- Privacy Policy
```
Is this related to Operations of the platform only ?

LCaywood replied:
Also relates to the operations of the community and to the privacy of the community. Need to confirm how Discourse & Discord's privacy policies would operate under a general privacy policy (likely adopted from the conservancy/foundation the Project may eventually be hosted by).

LCaywood replied:
Legal advisors have suggested incorporating the Privacy Policy within the Terms of Use.

olberger commented on governance.md:
```diff
@@ -0,0 +1,101 @@
+- Contributor License Agreement
+- Code of Conduct
+- Privacy Policy
+- Terms of Use
```
use of the platform as someone learning network automation, or use of the resources of the Project for committers ?

LCaywood replied:
Both

Commit:
Update Steering Committee membership

Amend Steering Committee description to include "minimally viable" and "preferred" compositions.

Commit:
Update governance.md

Mierdin committed:
Fix link spacing typo at end of gov doc

Signed-off-by: Matt Oswalt <matt@keepingitclassless.net>

Mierdin commented:
This was ratified on the weekly standup on October 22, 2019.
